cmd/dreamboat/main.go

```diff
@@ -227,11 +227,19 @@ func run() cli.ActionFunc {
 
 		timeRelayStart := time.Now()
 		as := &pkg.AtomicState{}
-		ds := datastore.NewDatastore(&datastore.TTLDatastoreBatcher{storage}, storage.DB)
+
+		hc := datastore.NewHeaderController()
```

Comment:
This seems something that should be created within the function `NewDatastore`, instead of out here. It is not used anywhere else except from inside the `Datastore`.

Reply:
Well, I disagree. As much as we can we should avoid any composition and initialization within constructors, as a rule of thumb. I'm happy to make it an interface within datastore.

Reply:
Okay,  good point.The thing is that the more components the relay has, the bigger the main will become. So, start extracting these initializations into functions, for readability. So in main we should have a function call for every major component initialization:
`initDatastore`: initializes badger, headerController and datastore. Returns datastore
`initRelay`: initializes processManager and relay. Returns relay

What do you think about this alternative?

Reply:
I would very much like to change this function, just didn't wanted to do this in this very PR :) 
As we can clean main up from using errgroup and other things we don't need. 
But for that there are some changes needed to Service and Becon that I think should be done prior to cleanup :)

Reply:
Perfect, then let's improve that in another PR